### PR TITLE
Helm chart: strengthen default security context

### PR DESCRIPTION
**Pull request by adutra**
This PR adds sane defaults for pod and container security contexts.

Since the default filesystem is now read-only, it also adds a new mount point for /tmp with read-write access.

It also adds two more options: `extraVolumes` and `extraVolumeMounts`. These can serve as escape hatches in case users need more locations with read-write access.

### CODE REVIEW

File: helm/nessie/templates/NOTES.txt [hunk 1]
```diff
@@ -15,6 +15,6 @@ To connect to Nessie, please execute the following commands:
   export SERVICE_IP=$(kubectl get svc --namespace {{ .Release.Namespace }} {{ include "nessie.fullname" . }} -o jsonpath="{ .status.loadBalancer.ingress[0].ip }")
   echo http://$SERVICE_IP:{{ index .Values.service.ports "nessie-server" }}
 {{- else if contains "ClusterIP" .Values.service.type }}
-  nohup kubectl --namespace {{ .Release.Namespace }} port-forward svc/nessie 19120:{{ index .Values.service.ports "nessie-server" }} &
+  nohup kubectl --namespace {{ .Release.Namespace }} port-forward svc/{{ include "nessie.fullname" . }} 19120:{{ index .Values.service.ports "nessie-server" }} &
```
Comment (adutra) [inline helm/nessie/templates/NOTES.txt:18]: Minor, unrelated fix.

**Review by dimas-b — DISMISSED** [review]
GH reports `no new line character at the end of file` in `values.yaml`... otherwise LGTM :+1:

**Comment by adutra** [comment]
Badk to draft, it probably needs to be rebased on top of #9456

**Comment by adutra** [comment]
Ready for  review, but the Helm tests will fail until there is an official image with the new UID and GID.

**Comment by adutra** [comment]
> Ready for review, but the Helm tests will fail until there is an official image with the new UID and GID.

Scratch that, I forgot that CI builds its own images for Helm testing.

**Review by dimas-b — APPROVED** [review]
